src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -15,9 +15,26 @@ public class Sections {
     @OneToMany(mappedBy = "line", cascade = CascadeType.ALL, orphanRemoval = true)
     private List<Section> sections = new ArrayList<>();
 
-    void addSection(Section newSection) {
-        validateSectionDistance(newSection);
+    public int size() { // TODO 오로지 테스트를 위한 메서드 이므로 삭제를 고려
+        return sections.size();
+    }
+
+    public List<Station> getStations() {
+        return sections.stream()
+                .flatMap(section -> Stream.of(section.getUpStation(), section.getDownStation()))
+                .distinct()
+                .collect(Collectors.toList());
+    }
 
+    public void deleteLastSection() {
+        if (!sections.isEmpty()) {
+            sections.remove(sections.size() - 1);
+            return;
+        }
+        throw new InvalidSectionException("삭제할 수 있는 지하철 구간이 없습니다.");
```

Comment:
Nit: 개인적인 취향 차이입니다만, 예외 케이스를 먼저 처리하는 방식도 고민해보시면 좋을 것 같아요.
```suggestion
        if (sections.isEmpty()) {
            throw new InvalidSectionException("삭제할 수 있는 지하철 구간이 없습니다.");
        }
        sections.remove(sections.size() - 1);
```